src/Squirrel/ReleasePackage.cs

```diff
@@ -164,21 +164,28 @@ public string CreateReleasePackage(string outputFile, string packagesRootDir = n
             }
         }
 
-        static Task extractZipWithEscaping(string zipFilePath, string outFolder)
+        public static Task extractZipWithEscaping(string zipFilePath, string outFolder, Action<int> progress)
         {
-			return Task.Run(() => {
-				Directory.CreateDirectory(outFolder);
+            return Task.Run(() => {
+                Directory.CreateDirectory(outFolder);
 
-				using (ZipFile zip = ZipFile.Read(zipFilePath)) {
-					foreach (ZipEntry entry in zip.Entries.ToList()) {
-						var parts = entry.FileName.Split('\\', '/').Select(x => Uri.UnescapeDataString(x));
-						var decoded = String.Join(Path.DirectorySeparatorChar.ToString(), parts);
-						entry.FileName = decoded;
-						entry.Extract(outFolder);
-					}
-				}
-			});
-		}
+                using (ZipFile zip = ZipFile.Read(zipFilePath)) {
+                    var totalItems = zip.Entries.Count;
+                    var currentItem = 1;
+
+                    foreach (ZipEntry entry in zip.Entries.ToList()) {
+                        var parts = entry.FileName.Split('\\', '/').Select(x => Uri.UnescapeDataString(x));
+                        var decoded = String.Join(Path.DirectorySeparatorChar.ToString(), parts);
+                        entry.FileName = decoded;
+                        entry.Extract(outFolder);
+
+                        var percentage = (currentItem * 100d) / totalItems;
```

Comment:
So now we also get progress updates even while extracting individual files?

Reply:
Yes we can. It is only in the _installPackageToAppDir_ function because it supports progress showing. There are also two other places where is this function used however without progress showing, just passing empty lambda function there because implementation of the progress showing would not be that easy there.